tests/libs/estdlib/test_gen_server.erl

```diff
@@ -158,6 +161,43 @@ wait_for(P) ->
         P -> ok
     end.
 
+test_timeout() ->
+    io:format("test_timeout\n"),
+    {ok, Pid} = gen_server:start(?MODULE, [], []),
+    0 = gen_server:call(Pid, get_num_timeouts),
+    ok = gen_server:call(Pid, {call_reply_timeout, 10}),
+    timer:sleep(11),
+    1 = gen_server:call(Pid, get_num_timeouts),
+    ok = gen_server:call(Pid, {call_reply_timeout, infinity}),
+    timer:sleep(11),
+    1 = gen_server:call(Pid, get_num_timeouts),
+    try
+        gen_server:call(Pid, {call_noreply_timeout, 10}, 100)
+    catch
+        _:_ -> ok
+    end,
+    2 = gen_server:call(Pid, get_num_timeouts),
+    ok = gen_server:cast(Pid, {cast_timeout, 10}),
+    timer:sleep(11),
+    3 = gen_server:call(Pid, get_num_timeouts),
+    ok = gen_server:cast(Pid, {cast_timeout, infinity}),
+    timer:sleep(11),
+    3 = gen_server:call(Pid, get_num_timeouts),
+    gen_server:cast(Pid, {set_info_timeout, 10}),
+    timer:sleep(11),
+    3 = gen_server:call(Pid, get_num_timeouts),
+    Pid ! ping,
+    timer:sleep(11),
+    N = gen_server:call(Pid, get_num_timeouts),
+    true = N > 3,
+    Pid ! ping,
+    timer:sleep(30),
+    M = gen_server:call(Pid, get_num_timeouts),
+    true = M > N + 1,
+    %%
+    gen_server:stop(Pid),
+    ok.
+
```

Comment:
The only thing I would ask is to break these up either into separate function with descriptive names, or add comments with a description and separate with whitespace.  Just makes it easier to read.

Reply:
Is it better like this?

Reply:
Yes that helps!